include/aws/common/uri.h

```diff
@@ -136,6 +136,28 @@ AWS_COMMON_API bool aws_uri_query_string_next_param(const struct aws_uri *uri, s
  */
 AWS_COMMON_API int aws_uri_query_string_params(const struct aws_uri *uri, struct aws_array_list *out_params);
 
+/**
+ * For iterating over the params in the query string.
+ * `param` is an in/out argument used to track progress, it MUST be zeroed out to start.
+ * If true is returned, `param` contains the value of the next param.
+ * If false is returned, there are no further params.
+ *
+ * Edge cases:
+ * 1) Entries without '=' sign are treated as having a key and no value.
+ *    Example: First param in query string "a&b=c" has key="a" value=""
+ *
+ * 2) Blank entries are skipped.
+ *    Example: The only param in query string "&&a=b" is key="a" value="b"
+ */
+AWS_COMMON_API bool aws_query_string_next_param(struct aws_byte_cursor query_string, struct aws_uri_param *param);
```

Comment:
in that [other PR](https://github.com/awslabs/aws-c-s3/pull/401/files#r1435315004) how do you plan to go from `path_and_query` to just `query`? Does that necessitate another helper?